src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -480,27 +478,28 @@ private static string GetPreferredUsernameFromIdToken(bool isAdfsAuthority, IdTo
                 return null;
             }
 
-            tokenCacheItems = FilterByHomeAccountTenantOrAssertion(requestParams, tokenCacheItems);
-            tokenCacheItems = FilterByTokenType(requestParams, tokenCacheItems);
-            tokenCacheItems = FilterByScopes(requestParams, tokenCacheItems);
-            tokenCacheItems = await FilterByEnvironmentAsync(requestParams, tokenCacheItems).ConfigureAwait(false);
+            FilterByHomeAccountTenantOrAssertion(requestParams, accessTokens);
```

Comment:
For these filtering methods, it is better to return the object that is being operated on to reveal which parameter the method is changing, rather than returning void. And or change the name to include the work "tokens" at least.